Sources/Valkey/Connection/ValkeyConnection.swift

```diff
@@ -189,8 +236,42 @@ public final actor ValkeyConnection: ValkeyClientProtocol, Sendable {
     public func execute<each Command: ValkeyCommand>(
         _ commands: repeat each Command
     ) async -> sending (repeat Result<(each Command).Response, Error>) {
+        #if DistributedTracingSupport
+        let span = startSpan("MULTI", ofKind: .client)
```

Comment:
Why `MULTI`? This implies you are running the Valkey command `MULTI` which starts a transaction.
Is it worthwhile building a string out of all of the commands eg `SET,LPUSH,INCR`